server/config_manager/config_manager.go

```diff
@@ -0,0 +1,509 @@
+	ErrDecode = func(e error) string {
+		return fmt.Sprintf("decode error: %v", e)
+	}
+	errNotSupported = "not supported"
+)
+
+// ConfigManager is used to manage all components' config.
+type ConfigManager struct {
+	sync.RWMutex
+	GlobalCfgs map[string]*GlobalConfig
+	LocalCfgs  map[string]map[string]*LocalConfig
+}
+
+// NewConfigManager creates a new ConfigManager.
+func NewConfigManager() *ConfigManager {
+	return &ConfigManager{
+		GlobalCfgs: make(map[string]*GlobalConfig),
+		LocalCfgs:  make(map[string]map[string]*LocalConfig),
+	}
+}
+
+// Persist saves the configuration to the storage.
+func (c *ConfigManager) Persist(storage *core.Storage) error {
+	err := storage.SaveComponentsConfig(c)
+	return err
+}
+
+// Reload reloads the configuration from the storage.
+func (c *ConfigManager) Reload(storage *core.Storage) error {
+	_, err := storage.LoadComponentsConfig(c)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *ConfigManager) getComponent(ComponentID string) string {
+	for component := range c.LocalCfgs {
+		for componentID := range c.LocalCfgs[component] {
+			if componentID == ComponentID {
+				return component
+			}
+		}
+	}
+	return ""
+}
+
+// Get returns config and the latest version.
+func (c *ConfigManager) Get(version *configpb.Version, component, componentID string) (*configpb.Version, string, *configpb.Status) {
+	c.RLock()
+	defer c.RUnlock()
+	var config string
+	var err error
+	var status *configpb.Status
+	if componentsCfg, ok := c.LocalCfgs[component]; ok {
+		if cfg, ok := componentsCfg[componentID]; ok {
+			config, err = c.getComponentCfg(component, componentID)
+			if err != nil {
+				return version, "", &configpb.Status{
+					Code:    configpb.StatusCode_UNKNOWN,
+					Message: ErrEncode(err),
+				}
+			}
+			if reflect.DeepEqual(cfg.getVersion(), version) {
+				status = &configpb.Status{Code: configpb.StatusCode_OK}
+			} else {
+				status = &configpb.Status{Code: configpb.StatusCode_WRONG_VERSION}
+			}
+		} else {
+			status = &configpb.Status{Code: configpb.StatusCode_COMPONENT_ID_NOT_FOUND}
+		}
+	} else {
+		status = &configpb.Status{Code: configpb.StatusCode_COMPONENT_NOT_FOUND}
+	}
+	return c.getLatestVersion(component, componentID), config, status
+}
+
+// Create is used for registering a component to PD.
+func (c *ConfigManager) Create(version *configpb.Version, component, componentID, cfg string) (*configpb.Version, string, *configpb.Status) {
+	c.Lock()
+	defer c.Unlock()
+	var status *configpb.Status
+	latestVersion := c.getLatestVersion(component, componentID)
+	initVersion := &configpb.Version{Local: 0, Global: 0}
+	if componentsCfg, ok := c.LocalCfgs[component]; ok {
+		if _, ok := componentsCfg[componentID]; ok {
+			// restart a component
+			if reflect.DeepEqual(initVersion, version) {
+				status = &configpb.Status{Code: configpb.StatusCode_OK}
+			} else {
+				status = &configpb.Status{Code: configpb.StatusCode_WRONG_VERSION}
+			}
```

Comment:
For restart component with default `Version` (i.e both local and global set to 0), we should expect that this component did not persist the `version` and `config` so we should return the latest version and config with `StatusCode_OK`.